config/initializers/service_providers.rb

```diff
@@ -0,0 +1,3 @@
+SERVICE_PROVIDERS = YAML.load_file("#{Rails.root}/config/service_providers.yml")
```

Comment:
Actually, that's not working. In order to test that the proper config is loaded based on the domain name, I need to be able to call a method to update the hash, since it's memoized. I wasn't able to get it to work with class variables. I'll merge this as is, and we can revisit later if there's a way to do it from the same class.